test/utils/structs/Checkpoints.test.js

```diff
@@ -75,70 +67,66 @@ contract('Checkpoints', function () {
 
         it('length', async function () {
-          expect(await this.methods.length()).to.be.bignumber.equal(this.checkpoints.length.toString());
+          expect(await this.methods.length()).to.equal(this.checkpoints.length);
         });
 
         it('returns latest value', async function () {
-          expect(await this.methods.latest()).to.be.bignumber.equal(last(this.checkpoints).value);
-
-          const ckpt = await this.methods.latestCheckpoint();
-          expect(ckpt[0]).to.be.equal(true);
-          expect(ckpt[1]).to.be.bignumber.equal(last(this.checkpoints).key);
-          expect(ckpt[2]).to.be.bignumber.equal(last(this.checkpoints).value);
+          const latest = this.checkpoints.at(-1);
+          expect(await this.methods.latest()).to.equal(latest.value);
+          expect(await this.methods.latestCheckpoint()).to.have.ordered.members([true, latest.key, latest.value]);
         });
 
         it('cannot push values in the past', async function () {
-          await expectRevertCustomError(
-            this.methods.push(last(this.checkpoints).key - 1, '0'),
+          await expect(this.methods.push(this.checkpoints.at(-1).key - 1n, 0n)).to.be.revertedWithCustomError(
+            this.mock,
             'CheckpointUnorderedInsertion',
-            [],
           );
         });
 
         it('can update last value', async function () {
-          const newValue = '42';
+          const newValue = 42n;
 
           // check length before the update
-          expect(await this.methods.length()).to.be.bignumber.equal(this.checkpoints.length.toString());
+          expect(await this.methods.length()).to.equal(this.checkpoints.length);
 
           // update last key
-          await this.methods.push(last(this.checkpoints).key, newValue);
-          expect(await this.methods.latest()).to.be.bignumber.equal(newValue);
+          await this.methods.push(this.checkpoints.at(-1).key, newValue);
+          expect(await this.methods.latest()).to.equal(newValue);
 
           // check that length did not change
-          expect(await this.methods.length()).to.be.bignumber.equal(this.checkpoints.length.toString());
+          expect(await this.methods.length()).to.equal(this.checkpoints.length);
         });
 
         it('lower lookup', async function () {
           for (let i = 0; i < 14; ++i) {
-            const value = first(this.checkpoints.filter(x => i <= x.key))?.value || '0';
+            const value = this.checkpoints.find(x => i <= x.key)?.value || 0n;
 
-            expect(await this.methods.lowerLookup(i)).to.be.bignumber.equal(value);
+            expect(await this.methods.lowerLookup(i)).to.equal(value);
           }
         });
 
         it('upper lookup & upperLookupRecent', async function () {
           for (let i = 0; i < 14; ++i) {
-            const value = last(this.checkpoints.filter(x => i >= x.key))?.value || '0';
+            const value = this.checkpoints.findLast(x => i >= x.key)?.value || 0n;
```

Comment:
Unfortunately `findLast` is a node 18 addition.
I'm tracking an update to Node in #4719